include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -168,12 +169,14 @@ inline void StatsdClient::send(const std::string& key,
                                    frequency);
     }
 
-    // Trim the trailing null chars
-    // TODO: perf improvement: send the len along and keep the buffer around as a member variable
-    buffer.resize(std::min(string_len, static_cast<int>(buffer.size())));
+    // Your stat was too large
+    if (static_cast<size_t>(string_len) > m_buffer.size()) {
+        m_sender->setErrorMessage("snprintf failed message too large");
```

Comment:
previously we would send the malformed message to statsd, now we error and tell the caller.

@vthiery i found this a bit messy since this class doesnt own the error message. do you have any better suggestions? we could make the client a friend class of the sender, that would work too and be less verbose but im not sure which you'd prefer.